pkg/ansible/run.go

```diff
@@ -24,36 +25,37 @@ import (
 	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
 	sdkVersion "github.com/operator-framework/operator-sdk/version"
 
-	log "github.com/sirupsen/logrus"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
+var log = logf.Log.WithName("ansible")
+
 func printVersion() {
-	log.Infof("Go Version: %s", runtime.Version())
-	log.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	log.Infof("Version of operator-sdk: %v", sdkVersion.Version)
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+	log.Info(fmt.Sprintf("Version of operator-sdk: %v", sdkVersion.Version))
 }
 
 // Run will start the ansible operator and proxy, blocking until one of them
 // returns.
-func Run(flags *aoflags.AnsibleOperatorFlags) {
+func Run(flags *aoflags.AnsibleOperatorFlags) error {
```

Comment:
@mhrivnak @shawn-hurley I'm uncertain whether the user would be better served if `Run()` returned an error vs. logging them. Thoughts? This discussion might relate to this [discussion](https://github.com/operator-framework/operator-sdk/pull/901#issuecomment-453664926) about when to have `cobra` handle the error vs. log the error.

Reply:
Good question. I think there are two cases to cover.

1. The operator failed to start and is exiting with some information to tell the user about what went wrong. Maybe the user needs to modify the runtime settings or the environment somehow. Maybe they just fat-fingered the command line.

2. The operator started fine and ran for a while, but eventually something went wrong that it couldn't recover from. In this case it's less likely that the user did something wrong with the CLI. Any feedback given is more likely related to a bug, or to some non-CLI-based occurrence (maybe the pod was killed, maybe it ran our of memory or file descriptors, etc.)

For case 1 we'd want very clear output on the command line about what happened. For 2 we also probably want that, but it's also important to make sure the failure is logged.

Looking at the "up local" code flow, if use case 2 happened, it would log a message that starts with "failed to run operator locally" even if the operator ran fine for hours. So that's an example where it assumes an error came from use case 1, but it actually was use case 2.

I know that's not a clear answer to your question, but I think nailing down these two cases may be the best way to get to an answer.